### DISCUSSION TITLE
Custom name isn't picked up on client side (codegen)

### DISCUSSION THREAD

**Discussion by RemyMachado** [Urigo/graphql-scalars#1789] [Q&A]
When I name my custom scalar, it is incorrectly picked up by [graphql-code-generator](https://github.com/dotansimha/graphql-code-generator) on the client side (via introspection query).

for `DateISO: GraphQLDateTime`, I end up with `DateTime` instead of my `DateISO` (custom name):
```typescript
// generated types (client)
export type Scalars = {
  DateTime: string // instead of DateISO: string
}
`
```

Note that the name is correctly generated on the server side codegen.
```typescript
// generated types (server)
export type Scalars = {
  DateISO: string // correct
}
```

Here's my implementation on the server side:
```typescript
// schemas
gql`
    scalar DateISO
`

// resolvers
const resolvers = [
    {
        DateISO: GraphQLDateTime,
    },
]
`
```

```yml
# codegen.yml
scalars:
        DateISO: string
```

Did I miss a step? Any help or clue would be greatly appreciated.
Thanks

N.B.
I'm not sure if I should ask this here or on the gql-codegen repo

**Comment by saihaj** [comment]
Can you share a [minimal reproduction](https://en.wikipedia.org/wiki/Minimal_working_example). You can either use [StackBlitz](https://stackblitz.com/), [CodeSandbox](https://codesandbox.io/) or a public GitHub repo.

**Reply by RemyMachado** [reply]
I'm afraid I can't. Since the `graphql-scalars` setup is pretty straight forward, I think it's a normal behavior (limitation), I doubt I exposed a bug with 3 lines of code. I wanted a simple feedback on the question. Has someone ever tried to use a different name? I'll stay with the original scalar's name `DateTime` in the meantime.

**Reply by saihaj** [reply]
if introspection is returning the `DateISO` then it should be working.
